### PR TITLE
vLLM-v0 upgrade: fixing multi-node training

### PR DESCRIPTION
**Pull request by ehsk**
A follow-up on #122 where vllm is upgraded from `v0.8.5.post1` to `v0.10.0`

The issue with multi-node training is fixed in this PR (see `_create_stateless_pg` in `torch_utils.py`), while other version upgrade changes are more or less the same as #122.

gray=`v0.8.5post1`, purple=`v0.10.0`, lightblue=`v0.10.0` on 2 nodes

| Logprobs | Reward |
|:--------:|:------:|
| Logprobs | Rewards per optimization steps |
 
 Full report is [here](https://wandb.ai/ehsk/watermelon/reports/ORZ-Qwen2-5-7B-vLLM-v0-upgrade-again---VmlldzoxNjU0MjQwMA).

### CODE REVIEW

File: pipelinerl/finetune_loop.py [hunk 1]
```diff
@@ -258,9 +259,10 @@ def send_weight_update(
                 logger.info(f"Published weight update request for version {version}")
                 for _, parameter in named_parameters.items():
                     # Use PyNcclCommunicator's broadcast method as torch.distributed does not work since vLLM disabled that transfer path
-                    # REVERTED: self.actor_update_group.broadcast(parameter.data, src=0, stream=torch.cuda.current_stream())
-                    dist.broadcast(parameter.data, src=0, group=self.actor_update_group)
-                dist.barrier(self.actor_update_group)
+                    self.actor_update_group.broadcast(parameter.data, src=0, stream=torch.cuda.current_stream())
+                    # Previously used torch.distributed.broadcast
+                    # dist.broadcast(parameter.data, src=0, group=self.actor_update_group)
```
Comment (rafapi) [inline pipelinerl/finetune_loop.py]: let's remove this line

File: pipelinerl/finetune_loop.py [hunk 2]
```diff
@@ -413,19 +415,19 @@ def run_finetuning_loop(
...
-            backend="nccl",
+        current_device = get_accelerator().device
+        torch.cuda.set_device(current_device)
+        logger.info("Initializing actor process group using StatelessProcessGroup")
+        logger.info(f"Set CUDA device to {current_device} for actor process group (rank 0)")
+        actor_update_group = torch_utils.stateless_init_process_group(
+        # logger.info("Initializing actor process group")
+        # actor_update_group = torch_utils.init_extra_process_group(
+            # group_name="actor",
+            # backend="nccl",
```
Comment (rafapi) [inline pipelinerl/finetune_loop.py]: let's remove the commented lines, i trust you :)

**Review by rafapi — APPROVED** [review]
LGTM!
